reframe/utility/lmod_helper.py

```diff
@@ -0,0 +1,67 @@
+import subprocess
+
+class PopenCommunicate():
```

Comment:
It would be great if this code could have a tighter integration with ReFrame.
The `PopenCommunicate()` should not be needed since ReFrame provides methods to run [external processes](
https://github.com/eth-cscs/reframe/blob/c9e3406c044843f570223f76d313912cc27795cc/reframe/utility/os_ext.py#L30).
Having said that, I am in the opinion that one needs to use the Python bindings for Lmod, which is the way that all module systems are implemented in ReFrame (https://github.com/eth-cscs/reframe/blob/master/reframe/core/modules.py). So, this class should be dropped.

Reply:
Hi @victorusu 
Thanks for the suggestions. I see your points. In fact, I pushed here what I had done in another context, so it's no surprise that it looks a bit like duct-taped ;-)

I'll try to get to your suggestions as soon as I have time.
Cheers,
Davide